server/src/main/java/org/opensearch/monitor/fs/FsHealthService.java

```diff
@@ -134,6 +146,9 @@ public StatusInfo getHealth() {
             statusInfo = new StatusInfo(HEALTHY, "health check disabled");
         } else if (brokenLock) {
             statusInfo = new StatusInfo(UNHEALTHY, "health check failed due to broken node lock");
+        } else if (lastRunStartTimeMillis.get() >= 0 && currentTimeMillisSupplier.getAsLong() -
+            lastRunStartTimeMillis.get() > healthyTimeoutThreshold.millis()) {
+            statusInfo = new StatusInfo(UNHEALTHY, "healthy threshold breached");
```

Comment:
Does it also need to account for ``REFRESH_INTERVAL_SETTING``? What if ``HEALTHY_TIMEOUT_SETTING``is  < ``REFRESH_INTERVAL_SETTING`` via settings update?

Reply:
Doesn't matter we want to know if the current time - start time of the current check exceeds the healthy threshold.

Reply:
@Bukhtawar I would like to disagree with you here, @getsaurabh02 has a good point, here is the reasoning:
 - the `lastRunStartTimeMillis` is only updated when REFRESH_INTERVAL_SETTING triggers (scheduled run)
 - the `REFRESH_INTERVAL_SETTING` should not exceed the `HEALTHY_TIMEOUT_SETTING` otherwise it would not be possible to distinguish if `flush()` operation is taking too long or the refresh is just not happening as often as it should

Alternatively, it is possible detach from the `REFRESH_INTERVAL_SETTING` completely and track `lastRunStartTimeMillis` / `lastRunStopTimeMillis`, so the duration of the flush operation is always known. Hope it makes sense, thank you.

Reply:
Thanks @reta for supporting this and adding your thoughts. 

I see one way to handle this is we reset ``lastRunStartTimeMillis`` to ``Long.MIN_VALUE`` post completion of every run of ``FsHealthMonitor``. That way the condition will be evalauted only for the current run.

**Scenario**
Here ``FsHealthMonitor`` is a Runnable which is scheduled to run at a fixed interval defined by  ``REFRESH_INTERVAL_SETTING``. For every execution, we are updating the class variable ``lastRunStartTimeMillis`` with the current timestamp at the start of run. This variable is however not updated post execution, and hence it will always have the start timestamp of the current (in-progress) or last (successful) run.

Since ``getHealth`` of ``NodeHealthService`` can be called independently anytime, such as during handling of ``FollowersChecker``, it will use the last updated value of ``lastRunStartTimeMillis`` to determine the healthy threshold breach. If previous successful run duration is beyond the healthy threshold, and current run is yet to start, this condition will end up being true always. @Bukhtawar Please correct me if I missing something here.

Reply:
@getsaurabh02 thank you, that would also work, I think the only drawback is that we are losing the `real` value of how long the FS check actually took. May not be important right now but I think it is very likely we would have to improve the detection algorithm to make the decisions based on the past 3-5 attempts (the other comment with retries you have had), the knowledge of the exact operation latency would be useful. In any case, could be revised later.

Reply:
Thanks for the catch!

Reply:
> that we are losing the real value of how long the FS check actually took
@reta That is covered as a part of slow logger that logs based on a slow logging threshold

Reply:
> I think it is very likely we would have to improve the detection algorithm to make the decisions based on the past 3-5 attempts

@reta The problem with this is what I mentioned earlier. Plus the chances that the node doesn't recover from the first freeze itself. In which case there won't be a second retry.

Reply:
@Bukhtawar correct, I think we may think about different way to approach that, but it is good as-is for the first cut, thank you

Reply:
@reta do you mind checking to see if changes look good to go. I pushed in the suggested fixes